Sources/Networking/Core/SampleResponseProvider.swift

```diff
@@ -0,0 +1,80 @@
+//
+//  SampleResponseProvider.swift
+//  
+//
+//  Created by Matej Molnár on 04.01.2023.
+//
+
+import Foundation
+
+// necessary for NSDataAsset import
+#if os(macOS)
+    import AppKit
+#else
+    import UIKit
+#endif
+
+// MARK: - SampleResponseProvider definition
+
+/// A response provider which creates responses for requests from corresponding data files stored in Assets.
+public class SampleResponseProvider: ResponseProviding {
```

Comment:
Let's make it `open` again, plus, can we rename it to `DefaultResponseProvider` or just `ResponseProvider`? I don't believe there should be anything called `sample` in the package itself

Reply:
About accessory levels I left some note. We may unify our approach overall. The rest I agree

Reply:
@schwarja @cejanen I think that `DefaultResponseProvider` is misleading since the default ResponseProvider is a `URLSession` instance. `ResponseProvider` is also not a great name because we may have multiple implementations of `ResponseProviding`. Maybe `AssetResponseProvider` or `AssetLoadingResponseProvider` would fit better. What do you think?

Reply:
`MockResponseProvider`?

Reply:
Makse sense 👍